testcases/OpTestSTB.py

```diff
@@ -0,0 +1,623 @@
+                                                    BMC_CONST.LOADED,
+                                                    BMC_CONST.NOT_LOADED)
+            if l_bk_loaded_status == BMC_CONST.LOADED:
+                print BMC_CONST.BOOTKERNEL_PAR_LOADED
+                return BMC_CONST.BOOTKERNEL_PAR_LOADED
+            elif l_bk_loaded_status == BMC_CONST.NOT_LOADED:
+                print BMC_CONST.BOOTKERNEL_PAR_NOT_LOADED
+                return BMC_CONST.BOOTKERNEL_PAR_NOT_LOADED
+            else:
+                print "BOOTKERNEL partition loaded related error occured"
+                return BMC_CONST.FW_FAILED
+        else:
+            l_msg = "BOOTKERNEL partition loaded details not seen"
+            raise OpTestError(l_msg)
+
+    ##
+    # @brief This function checks for the device-tree node of Trusted boot
+    # Returns FW_SUCCESS when TB_DT_ENTRY found
+    # Returns FW_FAILED when TB_DT_ENTRY not found
+    #
+    def proc_tb_verify(self):
+        l_cmd = "test -f %s; echo $?" % BMC_CONST.TB_DT_ENTRY
+        l_out = self.cv_HOST.host_run_command(l_cmd)
+        l_out = l_out.strip("\n\r")
+        if int(l_out) == 0:
+            print "Trusted Boot device-tree file exists"
+            return BMC_CONST.FW_SUCCESS
+        else:
+            print "No Trusted Boot device-tree file exists"
+            return BMC_CONST.FW_FAILED
+
+    ##
+    # @brief This function checks for the device-tree node of Secure boot
+    # Returns FW_SUCCESS when SB_DT_ENTRY found
+    # Returns FW_FAILED when SB_DT_ENTRY not found
+    #
+    def proc_sb_verify(self):
+        l_cmd = "test -f %s; echo $?" % BMC_CONST.SB_DT_ENTRY
+        l_out = self.cv_HOST.host_run_command(l_cmd)
+        l_out = l_out.strip("\n\r")
+        if int(l_out) == 0:
+            print "Secure Boot device-tree file exists"
+            return BMC_CONST.FW_SUCCESS
+        else:
+            print "No Secure Boot device-tree file exists"
+            return BMC_CONST.FW_FAILED
+
+    ##
+    # @brief This function cross verifies the output from hw_sb_support function
+    # which reads Secure boot value from SSR value and secure_boot_mode function
+    # which reads Secure boot value from OPAL messagelog
+    # When both the values are not same then raises OpTestError
+    #
+    def cross_check_hw_opal_sb(self):
+        if self.hw_sb_support() == self.secure_boot_mode():
+            print "For Secure boot Both SSR value and OPAL are in sync"
+        else:
+            l_msg = "For Secure boot Both SSR value and OPAL are NOT in sync"
+            print l_msg
+            raise OpTestError(l_msg)
+
+    ##
+    # @brief This function cross verifies the output from hw_tb_support function
+    # which reads Trusted boot value from SSR value and trusted_boot_mode
+    # function, which reads Trusted boot value from OPAL messagelog
+    # When both the values are not same then raises OpTestError
+    #
+    def cross_check_hw_opal_tb(self):
```

Comment:
you can follow above function verify_secure_boot_in_opal to write a new function verify_trusted_boot_in_opal for verifying trusted boot mode.